adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+      hbaseContext.hbaseRDD(tableName, scans)
+
+    }
+
+    def hbaseBulkLoad(genodata: RDD[(Array[Byte], List[(String, Array[Byte])])],
+                      hbaseTableName: String,
+                      flatMap: scala.Function1[(Array[Byte], List[(String, Array[Byte])]), scala.Iterator[scala.Tuple2[org.apache.hadoop.hbase.spark.KeyFamilyQualifier, scala.Array[scala.Byte]]]],
+                      stagingFolder: String) = {
+
+      val familyHBaseWriterOptions = new java.util.HashMap[Array[Byte], FamilyHFileWriteOptions]
+      val f1Options = new FamilyHFileWriteOptions("GZ", "ROW", 128, "FAST_DIFF")
+      familyHBaseWriterOptions.put(Bytes.toBytes("g"), f1Options)
+
+      // call implicit RDD function hbaseBulkLoad to write bulk files to staging locations
+      // and then load to HBase
+      genodata.hbaseBulkLoad(hbaseContext,
+        TableName.valueOf(hbaseTableName),
+        flatMap,
+        stagingFolder,
+        familyHBaseWriterOptions,
+        compactionExclude = false,
+        HConstants.DEFAULT_MAX_FILE_SIZE)
+
+      // This permission change appears necessary, plan to revisit to find better way
+      ("hadoop fs -chmod -R 660 " + stagingFolder) !
+
+      val load = new LoadIncrementalHFiles(conf)
+      load.doBulkLoad(new Path(stagingFolder), admin, connection.getTable(TableName.valueOf(hbaseTableName)),
+        connection.getRegionLocator(TableName.valueOf(hbaseTableName)))
+
+    }
+
+    def hbaseBulkDelete() {}
+
+  }
+
+  /**
+   * KeyStrategy abstract class
+   *
+   * @tparam T Type of rowKeyInfo class which is container for data needed to construct a row key
+   * @tparam U Type of rangePrefixInfo class which is a container for the data needed to define a key range
+   */
+  private[hbase] abstract class KeyStrategy[T, U] {
+    def getKey(rowKeyInfo: T): Array[Byte]
+    def getKeyRangePrefix(rangePrefixInfo: U): (Array[Byte], Array[Byte])
+  }
+
+  private[hbase] case class KeyStrategy1rowKeyInfo(contigName: String,
+                                                   start: Int,
+                                                   end: Int,
+                                                   refAllele: String,
+                                                   altAllele: String)
+
+  /**
+   * KeyStrategy1RangePrefixInfo contains a ReferenceRegion used to define a key range prefix in
+   * Keystrategy1
+   *
+   * @param queryRegion
+   */
+  private[hbase] case class KeyStrategy1RangePrefixInfo(queryRegion: ReferenceRegion)
+
+  /**
+   * KeyStrategy1  implements a HBase row key design consisting of underscore seperated fields:
+   *   contigName
+   *   start
+   *   refAlleel
+   *   AltAllele
+   *   length
+   *
+   */
+
+  private[hbase] object KeyStrategy1 extends KeyStrategy[KeyStrategy1rowKeyInfo, KeyStrategy1RangePrefixInfo] {
+    def getKey(rowKeyInfo: KeyStrategy1rowKeyInfo): Array[Byte] = {
+      Bytes.toBytes(rowKeyInfo.contigName + "_" +
```

Comment:
I would replace this with:
```
"%s_%10d_%s_%s_%d".format(rowKeyInfo.contigName,
  rowKeyInfo.start,
  rowKeyInfo.refAllele,
  rowKeyInfo.altAllele,
  rowKeyInfo.end - rowKeyInfo.start)
```

Also, do we need to handle nulls wrt altAllele? For gVCF reference model genotypes, we set alt to null.